### ISSUE TITLE
Punishment for those who constantly ignores QA

### ISSUE THREAD

**Issue by ypshenychka** [zerocracy/farm#1422]
We should have some kind of punishment for those users who constantly ignore QA remarks and pings.

Currently we have a "5 days rule", after which QA gives his verdict anyway. But it totally discourages some of the users to communicate with QA at all if they can wait 5 days for their payment and it's not a problem for them. But for QA it becomes a real "pain", when he has to track those tickets, ping users and wait, wait, wait in total ignorance.

I suggest to implement a punishment for users, who, let's say, didn't respond to QA after 5 days has passed in more than 2 tickets. It can be points deduction, for example. Not sure how this can be tracked easily (and automatically). Also we should pay attention that not always ticket performer has to make some changes and respond to QA. But how to track this "activity" for users who ignore QA, but are not assigned in this ticket?

Interesting to hear your thoughts about this situation.

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by ypshenychka** [comment]
@elenavolokhova What do you think? Does it make sense to implement something like this?

**Comment by g4s8** [comment]
@ypshenychka I think it make a sense but only for job performers, in case of open source project many people may comment some tickets but they may be not aware of our [rules](http://www.zerocracy.com/policy.html) and it's OK, I don't think we need to punish these people somehow. BTW, can'y you just say `quality bad` if the performer is not responding within 5 days?

**Comment by ypshenychka** [comment]
@g4s8 Sure, I can say "quality bad". But once or twice it can be forgiven, I assume (reasons for not repsonding can be different, but person should be paid for the work done). However if it's systematically (we should define how many "5 days" tickets were ignored previously), then we can say 'quality bad'. Also we should define strict rule - e.g. I'm waiting response from a person from team (who had previously once or twice ignored me already for 5 days, but I gave an objective verdict), but not performer of the ticket. Should I give bad verdict? It won't punish this person in any way.

Also how should we keep track of "stubborn violators"? It should be somehow stored in shared place, so that different QAs can see users statistics. E.g. @user already ignored @elenavolokhova for 5 days once in one of the projects, where I'm not working at. Now on my project he's ignoring me for 5 days as well. And it's for the first time for me this @user does this. But we should be counting this as second violation and, for example, give bad mark, or deduct points (whatever we choose).

**Comment by g4s8** [comment]
@ypshenychka maybe we can add new metric for such users, like "Negligence" for users who was rejected by 10-days delay often. @yegor256 what do you think?

**Comment by ypshenychka** [comment]
@g4s8 Also we should somehow take into consideration vacation. If a person doesn't repsond within 5 days, but he's on vacation, then don't count it towards "ignore".

**Comment by carlosmiranda** [comment]
@ypshenychka @g4s8 I think delayed payment is a sufficient disincentive for a performer. Do we have actual examples of team members being chronically negligent of QA? At least on my part I would prefer to fix issues sooner than later, because I want to be paid.

A bigger issue would be that some of our rules apply to non-team members who may not be aware of policy. See #1427 for example, which has bogged down my own tickets in the past. If an open-source project gets big enough QA/performer can't keep chasing non-team violators.

**Comment by ypshenychka** [comment]
@carlosmiranda I agree with #1427. I didn't have a lot of situations like this, but we can just ignore comments from non-team members. It's not a problem.

As per examples of such negligence, I have a bunch of such tickets. 
Here are the most recent ones that already passed QA: https://github.com/zold-io/zold/pull/362, https://github.com/zold-io/zold/pull/150, https://github.com/zold-io/zold/pull/152, https://github.com/zold-io/zold/pull/263, https://github.com/zold-io/zold/pull/273, https://github.com/zold-io/zold/pull/316, https://github.com/zold-io/zold/pull/335.

Some of them are currently 'in progress' - which means that I already forsee, that noone will respond to me after 5 days pass: https://github.com/zold-io/zold/pull/362, https://github.com/zold-io/zold/issues/322, https://github.com/zold-io/zold/issues/357, https://github.com/zold-io/zold/pull/223.

**Comment by g4s8** [comment]
@ypshenychka maybe you can ask for confirmation only in disputable situation when you can't be sure that quality is bad or not? but in all other cases just say the verdict without confirmation?

**Comment by carlosmiranda** [comment]
@ypshenychka agreed. I've "disputed" you a few times and managed to get the verdict updated from "acceptable" to "good" _at least once_. :) The notification process is very helpful.

**Comment by oltdaniel** [comment]
@ypshenychka I would like to add a vacation thing. Because I were on a vacation for 17 days, and were only able to respond to small things not to change or research information for updating comments.
